packages/flame/lib/src/collisions/quadtree/has_quadtree_collision_detection.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame/game.dart';
+
+/// Mixin should be applied to FlameGame to bring QuadTree collision support.
+/// [initCollisionDetection] should be called at [onLoad] function.
+///
+mixin HasQuadTreeCollisionDetection on FlameGame
+    implements HasCollisionDetection {
+  late CollisionDetection<ShapeHitbox> _collisionDetection;
+
+  @override
+  CollisionDetection<ShapeHitbox> get collisionDetection => _collisionDetection;
+
+  @override
+  set collisionDetection(CollisionDetection<Hitbox> cd) {
+    if (cd is! QuadTreeCollisionDetection) {
+      throw 'Must be QuadTreeCollisionDetection!';
+    }
+    _collisionDetection = cd;
+  }
+
+  /// Initialise QuadTree with params:
+  /// - [mapDimensions] describes collision area coordinates and size. Should
+  ///   match to game map position and size
+  /// - [maxObjects] (optional) - maximum objects count in one quadrant
+  /// - [maxLevels] (optional) - maximum nested quadrants
+  /// - [minimumDistance] (optional) - specify minimum distance between objects
+  ///   to consider them as possibly collideable. You also can reimplement
+  ///   [minimumDistanceCheck] if you need some custom behavior
+  ///
+  /// [broadPhaseCheck] checks if objects of different types should to collide.
+  /// The result of calculation is cached so you should not check here any
+  /// dynamical parameters, the function intended to be used as pure type
+  /// checker.
+  /// It usually should not be reimplemented, see
+  /// [CollisionCallbacks.broadPhaseCheck] instead
+  ///
+  void initCollisionDetection({
+    required Rect mapDimensions,
+    double? minimumDistance,
+    int maxObjects = 25,
+    int maxLevels = 10,
+  }) {
+    _collisionDetection = QuadTreeCollisionDetection(
+      mapDimensions: mapDimensions,
+      maxLevels: maxLevels,
+      maxObjects: maxObjects,
+      broadphaseCheck: broadPhaseCheck,
+      minimumDistanceCheck: minimumDistanceCheck,
+    );
+    this.minimumDistance = minimumDistance;
+  }
+
+  double? minimumDistance;
+
+  bool minimumDistanceCheck(Vector2 activeItemCenter, Vector2 potentialCenter) {
+    if (minimumDistance != null) {
+      if ((activeItemCenter.x - potentialCenter.x).abs() > minimumDistance! ||
+          (activeItemCenter.y - potentialCenter.y).abs() > minimumDistance!) {
+        return false;
+      } else {
+        return true;
+      }
+    }
+    return true;
```

Comment:
```suggestion
    return minimumDistance == null || 
       !((activeItemCenter.x - potentialCenter.x).abs() > minimumDistance! ||
         (activeItemCenter.y - potentialCenter.y).abs() > minimumDistance!);
```

This one will need formatting, the `!` could be removed too if the check within `()` is inverted.